### PR TITLE
Update troubleshooting guide, index and supported Kubernetes versions

### PR DESCRIPTION
**Pull request by LoreenRH**
### Proposed changes

I updated some parts of the docs such as:
 - changed --> for  → (to make them look prettier)
- Troubleshooting guide for Mac & Win (mostly diagnosing from terminal)
 - Make the install requirements clearer for D4W
 - Updated the smb permission parts (has changed in last stable & edge)
 - added information for the switching context menu for D4W 
 - Updated images according to latest release (18.06)
 - Added information for Kubernetes available in stable channel from 18.06 (both mac & windows)

### CODE REVIEW

**Comment by tfoxnc** [comment]
@LoreenRH Thanks so much for the updates. Our process is to have another set of eyes review big changes like this before merging. Do you have someone who can verify your changes? If so, please add them as a reviewer. Thanks again.

File: docker-for-windows/troubleshoot.md [hunk 1]
```diff
@@ -307,10 +340,9 @@ Here are some steps to take if you encounter similar problems:
     > Read the whole page before running any scripts, especially warnings about `-ForceDeleteAllSwitches`.
     > {: .warning}
 
-### Windows containers and Windows Server 2016
```
Comment (gtardif) [inline docker-for-windows/troubleshoot.md]: I think if we want to change this, do it in another PR, unrelated to Kubernetes.
Here we need to keep a mention of what users can expect on windows server. OK for your update on Windows 10. On windows server, we should mention Docker Desktop is not supported on windows Server and users can use docker EE basic that comes with Windows Server 2016. 
See Docker Desktop https://docs.docker.com/install/windows/docker-ee/#about-docker-ee-containers-and-windows-server
and http://success.docker.com/article/where-to-get-help-with-windows

File: docker-for-mac/troubleshoot.md [hunk 2]
```diff
@@ -25,21 +25,18 @@ GitHub](https://github.com/docker/for-mac/issues), or the [Docker for Mac
...
-results to Docker.  A diagnostic ID is generated, which must be provided when
-communicating with the Docker Team.  Optionally, you can open an issue on GitHub
-using the uploaded results and ID as a basis.
+Once the diagnostics are available, you can upload them and obtain a **Diagnostic ID**, which must be provided when communicating with the Docker team. For more information on our policy regarding personal data you can read [how is personal data handled in Docker Desktop](https://docs.docker.com/docker-for-mac/faqs/#how-is-personal-data-handled-in-docker-desktop).
 
 ![Diagnostics & Feedback with ID](images/diagnose-feedback-id.png){:width="600px"}
 
 If you click **Report an issue**, this opens [Docker for Mac issues on
 GitHub](https://github.com/docker/for-mac/issues/) in your web browser in a
-“create new issue” template, to be completed before submission.
```
Comment (gtardif) [inline docker-for-mac/troubleshoot.md]: We can add here : "do not forget to copy/paste your diagnostic ID".

File: docker-for-windows/troubleshoot.md [hunk 3]
```diff
@@ -7,18 +7,57 @@ redirect_from:
...
+**Diagnose & Feedback** from the menu bar.
+
+![Diagnose & Feedback](images/diagnose-feedback.png){:width="600px"}
+
+Once the **Diagnose & Feedback** window is opened, it will start to collect the dignostics. When the diagnostics are available, you can upload them and obtain a **Diagnostic ID**, which must be provided when communicating with the Docker team. For more information on our policy regarding personal data you can read [how is personal data handled in Docker Desktop](https://docs.docker.com/docker-for-mac/faqs/#how-is-personal-data-handled-in-docker-desktop).
+
+![Diagnose & Feedback with ID](images/diagnostic-id.png){:width="600px"}
+
+If you click on **Report an issue**, this opens [Docker for Windows issues on
+GitHub](https://github.com/docker/for-win/issues/) in your web browser in a "create new issue" template, to be completed before submision.
```
Comment (gtardif) [inline docker-for-windows/troubleshoot.md]: Also add here "Do not forget to copy/paste you diagnostic ID"

File: docker-for-windows/troubleshoot.md (hunk 1 above)
Comment (LoreenRH) [reply docker-for-windows/troubleshoot.md]: I took it off because in the text it only referred to an old link with With information on both.  But you are right I'll add a paragraph to link it to the windows server docs :D

File: docker-for-mac/troubleshoot.md (hunk 2 above)
Comment (LoreenRH) [reply docker-for-mac/troubleshoot.md]: @ahh-docker Fixed! ✨

**Comment by gtardif** [comment]
@ahh-docker are we waiting for more review on this ? (After looking at latest corrections, LGTM)

**Comment by chris-crone** [comment]
Fixes: https://github.com/docker/docker.github.io/issues/7172

**Comment by chris-crone** [comment]
We should also change `DOCKER_ORCHESTRATOR` to `DOCKER_STACK_ORCHESTRATOR`. We can do this in a followup though.

**Comment by gtardif** [comment]
@tfoxnc can you merge this doc update on Kubernetes ? or maybe @thaJeztah otherwise ?
(I'll fix the `DOCKER_STACK_ORCHESTRATOR` @chris-crone mentioned in a separate PR.)

**Comment by tfoxnc** [comment]
@chris-crone @gtardif What is the impact of not having the DOCKER_ORCHESTRATOR to DOCKER_STACK_ORCHESTRATOR change?

**Comment by chris-crone** [comment]
**tl;dr:** The feature doesn't work but the CLI currently prints out a warning if it finds `DOCKER_ORCHESTRATOR` instead of `DOCKER_STACK_ORCHESTRATOR` to inform users of the change.

Some background: prior to 18.03 EE and 18.06 CE, one could deploy stacks to Kubernetes using the experimental CLI flag plus `DOCKER_ORCHESTRATOR=kubernetes`. With these releases, the feature was moved to stable but the env var was changed to `DOCKER_STACK_ORCHESTRATOR` as no other docker subcommands supported Kubernetes.

**Comment by ahh-docker** [comment]
@JustinINevill Ok to merge? LGTM.

File: docker-for-windows/troubleshoot.md [hunk 4]
```diff
@@ -7,18 +7,57 @@ redirect_from:
...
+
+### Diagnosing from the terminal
+
+On occasions it is useful to run the diagnostics yourself, for instance if Docker for Windows cannot start.
+
+First locate the `com.docker.diagnose`, that should be in `C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe`.
+
+To create *and upload*  diagnostics in Powershell, run:
+
+```sh
```
Comment (thaJeztah) [inline docker-for-windows/troubleshoot.md]: This should probably be `powershell`, not `sh`

Comment (thaJeztah) [reply docker-for-windows/troubleshoot.md]: Also (I'm not a windows user), is `> &` part of the command to run, or is that the prompt?

**Comment by thaJeztah** [comment]
@LoreenRH I noticed there's quite some images being added/modified in this PR. Given that this repository is already huge, could you replace those images with optimised versions? I ran the files through [ImageOptim](https://imageoptim.com/mac), which _losslessly_ optimises the images; doing so made them ~1.1MB smaller

Could you;

- Check if the new images _replace_ some of the existing images; if so, replace the existing ones
- Replace the new images with the optimised versions (I'll attach here as a ZIP)
- Squash the commits (this is important, otherwise the non-optimised versions still end up in Git's history, so still make the repository big)

Thanks!

[optimized.zip](https://github.com/docker/docker.github.io/files/2437410/optimized.zip)

**Comment by LoreenRH** [comment]
@thaJeztah thanks for the review, I'll address your comments later today. I've got one question, when you say squash the commits, you mean all the commits to one, right? (just wanted to make sure 😅 )

**Comment by thaJeztah** [comment]
> when you say squash the commits, you mean all the commits to one, right?

oh! didn't notice there were more than one already.

It's ok to keep some separate commits (if they are grouped logically); squashing them all down would be ok as well of course.

The main reason I added that comment, is to make sure that you update (amend) the commits in which you added the images.

Hope I'm writing it so that you understand what I mean; it's a bit similar to container copy-on-write filesystem layers; so _don't_ do this;

- commit foo: add windows images (adds 1MB to repository) 
- commit bar: add mac images (adds 2MB to repository)
- commit baz: update images with optimised versions (images are smaller, but adds 2MB to repository) 

The above would make the repository 1+2+2=5MB bigger (not taking Git being "smart" into account)


But instead, update the original commits

- (updated with optimised versions) commit foo: add windows images (adds 500k to repository) 
- (updated with optimised versions) commit bar: add mac images (adds 1.5MB to repository)

This would make the repository only 2MB bigger

File: docker-for-windows/troubleshoot.md (hunk 4 above)
Comment (LoreenRH) [reply docker-for-windows/troubleshoot.md]: it didn't seem clear, I've changed it to `PS C:\>` because the `&` is part of the command :D

File: docker-for-mac/index.md [hunk 5]
```diff
@@ -252,9 +252,7 @@ changes, click another preference tab, then choose to discard or not apply chang
 
 ### Kubernetes
 
-Docker for Mac 17.12 CE (and higher) includes a standalone Kubernetes server
-that runs on your Mac, so that you can test deploying your Docker workloads on
-Kubernetes.
+In Docker for Mac [17.12 Edge (mac45)](/docker-for-mac/edge-relese-notes/#docker-community-edition-17120-ce-mac45-2018-01-05) and higher, and [18.06 Stable (mac70)](/docker-for-mac/release-notes/#docker-community-edition-18060-ce-mac70-2018-07-25) and higher, a standalone Kubernetes server is included that runs on your Mac, so that you can test deploying your Docker workloads on Kubernetes.
```
[the remark this answers is not in the document]
Comment (LoreenRH) [reply docker-for-mac/index.md]: I'll just do the wrapping here and after a PR about the current versions can be done later on (i guess it would be suitable before the next stable desktop)

File: docker-for-mac/troubleshoot.md [hunk 6]
```diff
@@ -16,7 +16,7 @@ browse and log issues on GitHub, and find workarounds for known problems.
 Hub](http://success.docker.com/) for knowledge base articles, FAQs, and
 technical support for various subscription levels.
 
-## Diagnose problems, send feedback, and create GitHub issues
+## Diagnose problems, send feedback and create GitHub issues
```
[the remark this answers is not in the document]
Comment (LoreenRH) [reply docker-for-mac/troubleshoot.md]: did the same fix for the troubleshooting guide on windows

**Comment by LoreenRH** [comment]
squashed everything to one commit to avoid spending too much time on rebasing each group of images :) it should have all the comments above addressed 😄

**Comment by thaJeztah** [comment]
oh! looks like this needs a minor rebase now 😕

**Comment by LoreenRH** [comment]
Now it should be good to go 🌮 ✨

**Comment by thaJeztah** [comment]
Thanks!
